karabo_data/geometry2.py

```diff
@@ -1055,8 +1055,114 @@ def invert_xfel_lpd_geom(path_in, path_out):
                 fout[path] = -fin[path][:]
 
 
-if __name__ == '__main__':
-    geom = AGIPD_1MGeometry.from_quad_positions(
-        quad_pos=[(-525, 625), (-550, -10), (520, -160), (542.5, 475)]
-    )
-    geom.write_crystfel_geom('sample.geom')
+class DSSC_Geometry(DetectorGeometryBase):
+    """Detector layout for DSSC
+
+    The coordinates used in this class are 3D (x, y, z), and represent multiples
+    of the pixel size.
+
+    You won't normally instantiate this class directly:
+    use one of the constructor class methods to create or load a geometry.
+    """
+    # Hexagonal pixels, 236 μm step in slow-scan axis, 204 μm in fast-scan
```

Comment:
So prefer approximating the hexagons to squares? We could also assume that the pixels are 236x272 (the pointy bit inclusive)

Reply:
I'm still unsure about exactly how to represent it in the geometry machinery. But if we are approximate them as rectangular pixels for drawing, the important dimension is really the step distance from one pixel to the next, rather than the size across an individual pixel. That makes it easy to calculate an approximately accurate position for each pixel.